blocks/bricks/base.py

```diff
@@ -327,6 +328,16 @@ def copy_and_tag(variable, role, name):
         return unpack(outputs)
 
 
+# Ensure that Application instance are not pickled, and that when
+# unpickling, a reference to an existing Application object is
+# substituted instead of creating a new one.
```

Comment:
Sorry, I misunderstood the original issue. But this comment is extremely confusing.